sequencer/src/block/full_payload/ns_table.rs

```diff
@@ -306,6 +286,57 @@ impl NsTable {
             index.0 * (NS_ID_BYTE_LEN + NS_OFFSET_BYTE_LEN) + NUM_NSS_BYTE_LEN + NS_ID_BYTE_LEN;
         usize_from_bytes::<NS_OFFSET_BYTE_LEN>(&self.bytes[start..start + NS_OFFSET_BYTE_LEN])
     }
+
+    /// Helper for [`NsTable::validate`], used in our custom [`serde`]
+    /// implementation.
+    ///
+    /// Checks conditions 1-3 of [`NsTable::validate`]. Those conditions can be
+    /// checked by looking only at the contents of the [`NsTable`].
+    fn validate_deserialization_invariants(&self) -> Result<(), NsTableValidationError> {
+        use NsTableValidationError::*;
+
+        // Byte length for a table with `x` entries must be exactly
+        // `x * NsTableBuilder::entry_byte_len() + NsTableBuilder::header_byte_len()`
```

Comment:
What is described in the comment and the condition of the `if` statement do not seem equivalent.

Reply:
I believe they are equivalent. Explainer comment added in https://github.com/EspressoSystems/espresso-sequencer/pull/1642/commits/50c2abcf2176573b0184cafd834e9b50c03fb870.

Reply:
Thanks!